pkg/mq/msgdispatcher/manager.go

```diff
@@ -190,87 +163,161 @@ func (c *dispatcherManager) Run() {
 		case <-ticker1.C:
 			c.uploadMetric()
 		case <-ticker2.C:
+			c.buildDispatcher()
 			c.tryMerge()
-		case <-c.lagNotifyChan:
-			c.mu.Lock()
-			c.lagTargets.Range(func(vchannel string, t *target) bool {
-				c.split(t)
-				c.lagTargets.GetAndRemove(vchannel)
-				return true
-			})
-			c.mu.Unlock()
 		}
 	}
 }
 
-func (c *dispatcherManager) tryMerge() {
-	log := log.With(zap.String("role", c.role), zap.Int64("nodeID", c.nodeID))
+func (c *dispatcherManager) buildDispatcher() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```

Comment:
Is this lock locked too early?
Try not to hold locks for IO

Reply:
This lock is not acquired too early, as it is immediately followed by reading mainDispatcher and deputyDispatcher.